src/rust/engine/src/tasks.rs

```diff
@@ -99,6 +99,11 @@ impl Tasks {
         product: types.files_content,
         input: types.directory_digest,
       },
+      Intrinsic {
+        kind: IntrinsicKind::DirectoryDigest,
```

Comment:
Calling this `MergedDirectories` would be a bit more clear I think.

Reply:
Removed `kind` entirely; it currently duplicates `product`. This way, we don't have the weird "you may try to project the wrong input" issue, too.

Happy to revert this if you prefer the exhaustiveness checking of the enum